devicetypes/zooz/zooz-zse11-q-sensor.src/zooz-zse11-q-sensor.groovy

```diff
@@ -0,0 +1,439 @@
+		capability "Tamper Alert"
+		capability "Temperature Measurement"
+		capability "Illuminance Measurement"
+		capability "Relative Humidity Measurement"
+		capability "Battery"
+		capability "Refresh"
+		capability "Health Check"
+		capability "Power Source"
+
+		// zw:Ss2 type:0701 mfr:027A prod:0200 model:0006 ver:1.09 zwv:6.04 lib:03 cc:5E,6C,55,98,9F sec:86,72,71,59,85,80,84,73,30,31,70,5A,7A
+		fingerprint mfr:"027A", prod:"0200", model:"0006", deviceJoinName: "Zooz Multipurpose Sensor" // Zooz ZSE11 Q Sensor (EU)
+		// zw:Ss2 type:0701 mfr:027A prod:0201 model:0006 ver:1.09 zwv:6.04 lib:03 cc:5E,6C,55,98,9F sec:86,72,71,59,85,80,84,73,30,31,70,5A,7A
+		fingerprint mfr:"027A", prod:"0201", model:"0006", deviceJoinName: "Zooz Multipurpose Sensor" // Zooz ZSE11 Q Sensor (US)
+		// zw:Ss2 type:0701 mfr:027A prod:0202 model:0006 ver:1.09 zwv:6.04 lib:03 cc:5E,6C,55,98,9F sec:86,72,71,59,85,80,84,73,30,31,70,5A,7A
+		fingerprint mfr:"027A", prod:"0202", model:"0006", deviceJoinName: "Zooz Multipurpose Sensor" // Zooz ZSE11 Q Sensor (AU)
+	}
+
+	preferences {
+		configParams.each { param ->
+			if (param.options) {
+				input "configParam${param.num}", "enum",
+					title: "${param.name}:",
+					required: false,
+					displayDuringSetup: false,
+					defaultValue: param.defaultVal,
+					options: param.options
+			} else if (param.range) {
+				input "configParam${param.num}", "number",
+					title: "${param.name}:",
+					required: false,
+					displayDuringSetup: false,
+					defaultValue: param.defaultVal,
+					range: param.range
+			}
+		}
+
+		input "tempOffset", "decimal",
+			title: "Temperature Offset:",
+			required: false,
+			defaultValue: 0,
+			range: "-50..50"
+
+		input "humidityOffset", "number",
+			title: "Humidity Offset:",
+			required: false,
+			defaultValue: 0,
+			range: "-50..50"
+
+		input "lightOffset", "number",
+			title: "Light Offset:",
+			required: false,
+			defaultValue: 0,
+			range: "-20000..20000"
+	}
+}
+
+def installed() {
+	log.debug "installed()..."
+	state.firstConfig = true
+	initialize()
+}
+
+def updated() {
+	log.debug "updated()..."
+
+	initialize()
+
+	if (!state.firstConfig) {
+		if (device.currentValue("powerSource") == "battery") {
+			logForceWakeupMessage("Configuration changes will be sent to the device the next time it wakes up.")
+		} else {
+			sendHubCommand(getConfigCmds())
+		}
+	} else {
+		sendHubCommand(getRefreshCmds())
+		state.firstConfig = false
+	}
+}
+
+void initialize() {
+	if (!device.currentValue("checkInterval")) {
+		sendEvent(name: "checkInterval", value: ((60 * 60 * 24) + (60 * 5)), displayed: false, data:[protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	}
+
+	if (!device.currentValue("tamper")) {
+		sendEvent(name: "tamper", value: "clear")
+	}
+
+	if (device.currentValue("powerSource") == null) {
+		boolean hasBatteryCC = ((zwaveInfo?.cc?.find { it.toString() == batteryCC }) || (zwaveInfo?.sec?.find { it.toString() == batteryCC })) 
```

Comment:
Does the device change its reported command classes after being added? It looks like all of your fingerprints have the battery command class.

Reply:
Yes, the command classes are different based on whether or not the device is plugged in when you join it.
I happened to have it joined on batteries when I copied the description and then just changed the product in it for the other 2 fingerprints since there just for different regions.

Do I need to include the raw description for both ways it can join?   I've never seen any of the built-in handlers do that before...

Reply:
Technically, I think the raw description would also be different if secure inclusion failed so technically to make that statement match the raw description list I'd have to add 4 comments...

Reply:
Okay. So the command classes are consistent once the device joins. That's all I wanted to know. The `zwaveInfo` object doesn't get updated after join and I just wanted to make sure you weren't expecting it.

Reply:
correct.  Let me know if this PR is all set now.  Thanks